Content.Shared/Interfaces/GameObjects/Components/Interaction/AttackEvent.cs

```diff
@@ -51,15 +32,46 @@ public class AttackEvent : EntityEventArgs
         /// </summary>
         public IEntity? TargetEntity { get; }
 
-        public AttackEvent(IEntity user, EntityCoordinates clickLocation, bool wideAttack, EntityUid target = default)
+        /// <summary>
+        ///     Modified by the handler to indicate whether the attack succeeded.
+        /// </summary>
+        public bool Succeeded { get; set;  }
+
+        public ClickAttackEvent(IEntity user, EntityCoordinates clickLocation, EntityUid target = default)
         {
             User = user;
             ClickLocation = clickLocation;
-            WideAttack = wideAttack;
             Target = target;
 
             IoCManager.Resolve<IEntityManager>().TryGetEntity(Target, out var targetEntity);
             TargetEntity = targetEntity;
         }
     }
+
+    /// <summary>
+    ///     Raised when a target entity is wide attacked by a user.
+    /// </summary>
+    public class WideAttackEvent : EntityEventArgs
+    {
+        /// <summary>
+        ///     Entity that triggered the attack.
+        /// </summary>
+        public IEntity User { get; }
+
+        /// <summary>
+        ///     The original location that was clicked by the user.
+        /// </summary>
+        public EntityCoordinates ClickLocation { get; }
+
+        /// <summary>
+        ///     Modified by the handler to indicate whether the attack succeeded.
+        /// </summary>
+        public bool Succeeded { get; set;  }
+
+        public WideAttackEvent(IEntity user, EntityCoordinates clickLocation)
```

Comment:
Same as above, add the entity being used to attack so this can be converted to a broadcasted event nicely